backend/RestAPI/Controllers/VideosController.cs

```diff
@@ -438,5 +438,43 @@ public async Task<IActionResult> DeleteVideos([FromBody] List<Guid> ids)
 
             return Ok();
         }
+
+        [HttpGet, Route("Recycled")]
+        public async Task<IActionResult> AllRecycledVideos()
+        {
+            var userIdClaim = User.FindFirst(x => x.Type == ClaimTypes.NameIdentifier);
+            if (userIdClaim is null)
+            {
+                return NotFound();
+            }
+
+            if (!Guid.TryParse(userIdClaim.Value, out var userId))
+            {
+                return NotFound();
+            }
+
+            var user = await _usersRepository.GetUserById(userId);
+            if (user is null) 
+            { 
+                return NotFound(); 
+            }
+
+            var videos = await _videosRepository.GetDeletedVideosByUserId(userId);
+            List<DeletedVideoDto> videoDtos = new List<DeletedVideoDto>();
+
+            foreach(var video in videos)
+            {
+                videoDtos.Add(new DeletedVideoDto()
+                {
+                    Id = video.Id,
+                    Title = video.Title,
+                    Size = video.Size,
+                    UploadDate = video.UploadDate.ToString("yyyy-MM-dd"),
+                    DeleteDate = video.DeleteDate?.ToString("yyyy-MM-dd")
```

Comment:
I approved this PR, but still have one question. Do we really need this field in front-end if it will only be used by service worker in backend, which deletes videos? @AArlauskas  @kibartas  @salmonela7

Reply:
I added it in case we are planning to sort the videos in recycle bin by deletion date.

Reply:
We talked, yesterday, that videos will be sorted by DeletionDate in bin, bro

Reply:
Yes, but Aurimas didn't change the `DeleteDate` field to reflect when the video was deleted. This field shows when the video will be deleted by yet to be implemented service worker(a month after the user marks this video for deletion). So it willl be really weird for the user to see dates in front-end which are in future.